app/src/org/commcare/dalvik/activities/ReportProblemActivity.java

```diff
@@ -40,21 +40,48 @@ public void onClick(View v) {
         sendReportEmail(reportEntry);
         finish();
     }
-    
-    public String buildMessage(String userInput){
-        
-        SharedPreferences prefs = CommCareApplication._().getCurrentApp().getAppPreferences();
-        
-        String username;
+
+    public static String getDomain(){
+        try {
+            SharedPreferences prefs = CommCareApplication._().getCurrentApp().getAppPreferences();
+            return prefs.getString(HttpRequestGenerator.USER_DOMAIN_SUFFIX, "not found");
+        } catch(NullPointerException e){
+            return "Domain not set.";
+        }
+    }
+
+    public static String getPostURL(){
+        try{
+            SharedPreferences prefs = CommCareApplication._().getCurrentApp().getAppPreferences();
+            return prefs.getString(HttpRequestGenerator.USER_DOMAIN_SUFFIX, "not found");
+        } catch(NullPointerException e){
+            return "PostURL not set.";
```

Comment:
Would be good to be more explicit about error handling.
Maybe:
CommCareApp currentApp = CommCareApplication._().getCurrentApp();
if (currentApp == null) {
    return "No installed apps";
}
return currentApp.getAppPreferences().getString(HttpRequestGenerator.USER_DOMAIN_SUFFIX, "PostUrl not found");